src/docfx/build/metadata/ContributionProvider.cs

```diff
@@ -210,4 +219,21 @@ private DateTime GetUpdatedAt(FilePath file, Repository? repository, GitCommit[]
             ? _input.GetLastWriteTimeUtc(file)
             : default;
     }
+
+    private static string GetGitHubUsernameWhenCommitWithNoReplyEmail(string email)
```

Comment:
Prefer `null` over empty string to communicate this method may not always succeed.
Is the first number github id? if so, populate it to `GitHubUser.Id` field

```suggestion
    private static GitHubUser? TryGetGitHubUserFromNoReplyEmail(string email)
```